rust_src/src/buffers.rs

```diff
@@ -232,7 +232,8 @@ impl LispBufferRef {
     }
 
     pub fn markers(self) -> Option<LispMarkerRef> {
-        unsafe { (*self.text).markers.as_ref().map(|m| mem::transmute(m)) }
+        let markers = unsafe { (*self.text).markers };
+        LispMarkerRef::from_ptr(markers as *mut _)
```

Comment:
Could we add a macro to reduce the noise required for these conversions? Bonus points for a way to discover the type.
```
macro_rules! SOME_GREAT_NAME {
    ($kind:expr, $symbol:expr) => {
        $kind::from_ptr($symbol as *mut _)
    }
}

SOME_GREAT_NAME!(LispMarkerRef, markers)
```

Reply:
rust 1.38 introduced a `cast` method for pointers which would make the `as *mut _` slightly nicer (the impl is basically just that but generic).

With that this would look like `LispMarkerRef::from_ptr(markers.cast())`. I suppose we can follow that name and have something like `markers.cast()` perform this itself as these are all `ExternalPtr<T>`

Reply:
I managed to implement something like this for different `ExternalPtr<T>` which I think looks pretty neat, but this call and a few others are specifically arbitrary pointers which gets a bit trickier. I apparently cannot implement `impl<T> From<*mut c_void> for Option<ExternalPtr<T>>` because `Option` isn't ours (I think that's what the compilers is telling me).

I can wrap it and implement `impl<T> From<*mut c_void> for MaybeExternalPtr<T>` but now I have one type I don't really want. Maybe this will work in combination with a macro or maybe `Deref`?